azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -696,8 +705,13 @@ private Status runJob() {
         logError("Job run failed, but will treat it like success.");
         logError(e.getMessage() + " cause: " + e.getCause(), e);
       } else {
-        finalStatus = changeStatus(Status.FAILED);
-        logError("Job run failed!", e);
+        if (node.getStatus() == Status.KILLED) {
```

Comment:
With this change, does the code in run method azkaban.execapp.JobRunner#run still apply?

    if (isKilled()) {
      // even if it's killed, there is a chance that the job failed is marked as
      // failure,
      // So we set it to KILLED to make sure we know that we forced kill it
      // rather than
      // it being a legitimate failure.
      finalStatus = changeStatus(Status.KILLED);
    }